lib/ebpd_loader.h

```diff
@@ -0,0 +1,14 @@
+#ifndef __EBPD_LOADER_H_
+#define __EBPD_LOADER_H_
+
+class EbpdLoader {
+    public:
+        EbpdLoader() { }
+        ~EbpdLoader();
+        int LoadFile(const char *filepath, int ifindex);
+        int LoadBuffer(void *buf, int buf_size, const char *name);
+    private:
+        void *handle;
```

Comment:
void* handle = nullptr. Otherwise your check "if(handle)" in ~EbpdLoader() can read uninitialized data.

Even better would be std::unique_ptr<void> with a custom deleter:

struct HandleDtor {
    void operator()(const void* handle) { ebpd_unload(handle); }
};
using UniqueHandle = std::unqiue_ptr<void, HandleDtor>;

Reply:
I also recommend using std::unique_ptr. It should have no overhead, and makes the code safer.
(eg, LoadFile was copiable above, which if done accidentally would have introduced an hard to find / fix bug, unique_ptr provides move semantics).

If we have something like:

using XdpHandle = ... ; // like Joel describred.

We can change LoadFile and LoadBuffer to be free form functions:

  XdpHandle LoadXdpFile(const std::string_view path, int ifindex);
  XdpHandle LoadXdpBuffer(const std::string_view buffer, const std::string_view name, int ifindex);

Where XdpHandle will be nullptr / false if program cannot be loaded. 
By using std::unique_ptr, the object becomes un-copiable.

Reply:
Yes, using unqiue_ptr in the new diff. Please take a look at my new pull request #22.